prov/usnic/src/usnic_direct/usnic_verbs_kern-abi.h

```diff
@@ -0,0 +1,882 @@
+struct ibv_destroy_srq {
+	__u32 command;
+	__u16 in_words;
+	__u16 out_words;
+	__u64 response;
+	__u32 srq_handle;
+	__u32 reserved;
+};
+
+struct ibv_destroy_srq_resp {
+	__u32 events_reported;
+};
+
+/*
+ * Compatibility with older ABI versions
+ */
+
+enum {
+	IB_USER_VERBS_CMD_QUERY_PARAMS_V2,
+	IB_USER_VERBS_CMD_GET_CONTEXT_V2,
+	IB_USER_VERBS_CMD_QUERY_DEVICE_V2,
+	IB_USER_VERBS_CMD_QUERY_PORT_V2,
+	IB_USER_VERBS_CMD_QUERY_GID_V2,
+	IB_USER_VERBS_CMD_QUERY_PKEY_V2,
+	IB_USER_VERBS_CMD_ALLOC_PD_V2,
+	IB_USER_VERBS_CMD_DEALLOC_PD_V2,
+	IB_USER_VERBS_CMD_CREATE_AH_V2,
+	IB_USER_VERBS_CMD_MODIFY_AH_V2,
+	IB_USER_VERBS_CMD_QUERY_AH_V2,
+	IB_USER_VERBS_CMD_DESTROY_AH_V2,
+	IB_USER_VERBS_CMD_REG_MR_V2,
+	IB_USER_VERBS_CMD_REG_SMR_V2,
+	IB_USER_VERBS_CMD_REREG_MR_V2,
+	IB_USER_VERBS_CMD_QUERY_MR_V2,
+	IB_USER_VERBS_CMD_DEREG_MR_V2,
+	IB_USER_VERBS_CMD_ALLOC_MW_V2,
+	IB_USER_VERBS_CMD_BIND_MW_V2,
+	IB_USER_VERBS_CMD_DEALLOC_MW_V2,
+	IB_USER_VERBS_CMD_CREATE_CQ_V2,
+	IB_USER_VERBS_CMD_RESIZE_CQ_V2,
+	IB_USER_VERBS_CMD_DESTROY_CQ_V2,
+	IB_USER_VERBS_CMD_POLL_CQ_V2,
+	IB_USER_VERBS_CMD_PEEK_CQ_V2,
+	IB_USER_VERBS_CMD_REQ_NOTIFY_CQ_V2,
+	IB_USER_VERBS_CMD_CREATE_QP_V2,
+	IB_USER_VERBS_CMD_QUERY_QP_V2,
+	IB_USER_VERBS_CMD_MODIFY_QP_V2,
+	IB_USER_VERBS_CMD_DESTROY_QP_V2,
+	IB_USER_VERBS_CMD_POST_SEND_V2,
+	IB_USER_VERBS_CMD_POST_RECV_V2,
+	IB_USER_VERBS_CMD_ATTACH_MCAST_V2,
+	IB_USER_VERBS_CMD_DETACH_MCAST_V2,
+	IB_USER_VERBS_CMD_CREATE_SRQ_V2,
+	IB_USER_VERBS_CMD_MODIFY_SRQ_V2,
+	IB_USER_VERBS_CMD_QUERY_SRQ_V2,
+	IB_USER_VERBS_CMD_DESTROY_SRQ_V2,
+	IB_USER_VERBS_CMD_POST_SRQ_RECV_V2,
+	/*
+	 * Set commands that didn't exist to -1 so our compile-time
+	 * trick opcodes in IBV_INIT_CMD() doesn't break.
+	 */
+	IB_USER_VERBS_CMD_CREATE_COMP_CHANNEL_V2 = -1,
+};
+
+struct ibv_destroy_cq_v1 {
+	__u32 command;
+	__u16 in_words;
+	__u16 out_words;
+	__u32 cq_handle;
+};
+
+struct ibv_destroy_qp_v1 {
+	__u32 command;
+	__u16 in_words;
+	__u16 out_words;
+	__u32 qp_handle;
+};
+
+struct ibv_destroy_srq_v1 {
+	__u32 command;
+	__u16 in_words;
+	__u16 out_words;
+	__u32 srq_handle;
+};
+
+struct ibv_get_context_v2 {
+	__u32 command;
+	__u16 in_words;
+	__u16 out_words;
+	__u64 response;
+	__u64 cq_fd_tab;
+	__u64 driver_data[0];
+};
+
+struct ibv_create_cq_v2 {
+	__u32 command;
+	__u16 in_words;
+	__u16 out_words;
+	__u64 response;
+	__u64 user_handle;
+	__u32 cqe;
+	__u32 event_handler;
+	__u64 driver_data[0];
+};
+
+struct ibv_modify_srq_v3 {
+	__u32 command;
+	__u16 in_words;
+	__u16 out_words;
+	__u32 srq_handle;
+	__u32 attr_mask;
+	__u32 max_wr;
+	__u32 max_sge;
+	__u32 srq_limit;
+	__u32 reserved;
+	__u64 driver_data[0];
+};
+
+struct ibv_create_qp_resp_v3 {
+	__u32 qp_handle;
+	__u32 qpn;
+};
+
+struct ibv_create_qp_resp_v4 {
+	__u32 qp_handle;
+	__u32 qpn;
+	__u32 max_send_wr;
+	__u32 max_recv_wr;
+	__u32 max_send_sge;
+	__u32 max_recv_sge;
+	__u32 max_inline_data;
+};
+
+struct ibv_create_srq_resp_v5 {
+	__u32 srq_handle;
+};
+
+#endif /* KERN_ABI_H */
```

Comment:
Do you want to keep all definitions, or limit to only those structures that you need?